core/src/main/scala/algebra/ring/Ring.scala

```diff
@@ -32,11 +32,11 @@ trait Ring[@sp(Int, Long, Float, Double) A] extends Any with Rig[A] with Rng[A]
   def fromInt(n: Int): A = sumN(one, n)
 }
 
-trait RingFunctions {
+trait RingFunctions extends AdditiveGroupFunctions with MultiplicativeMonoidFunctions {
```

Comment:
I wonder if these should be parameterized
```scala
trait RingFunctions[R[T] <: Ring[T]] {
  def fromInt[A](n: Int)(implicit ev: R[A]): A = ...
}
```
I'm worried there is some corner case of implicit resolution such that: `CommutativeRing[T].plus(a, b)` and `CommutativeRing.plus(a, b)` resolve to different implicits (one finds a `CommutativeRing[Int]` the other finds a `Ring[Int]`, which are not the same).